MergerService/Utils/TaskUtils.cs

```diff
@@ -1,46 +1,172 @@
+                }
+
+                try
+                {
+                    return JsonConvert.DeserializeObject<MergeTask>(taskData, this._jsonSerializerSettings)!;
+                }
+                catch (Exception e)
+                {
+                    this._logger.LogWarning(e, "Error deserializing returned task");
+                    return null;
+                }
+            }
+
+            // TODO: add heartbeat stop method
+        }
+
+        public void NotifyOnCompletion(string jobId, string taskId)
+        {
+            using (this._activitySource.StartActivity("notify overseer on task completion"))
+            {
+                // Notify overseer on task completion
+                this._logger.LogInformation($"Notifying overseer on completion, job: {jobId}, task: {taskId}");
+                string relativeUri = $"tasks/{jobId}/{taskId}/completed";
+                string url = new Uri(new Uri(_overseerUrl), relativeUri).ToString();
+                _ = this._httpClient.PostDataString(url, null);
+            }
+        }
+
+        private void Update(string jobId, string taskId, HttpContent content)
+        {
+            string relativeUri = $"jobs/{jobId}/tasks/{taskId}";
+            string url = new Uri(new Uri(_jobManagerUrl), relativeUri).ToString();
+            _ = this._httpClient.PutDataString(url, content);
+        }
+
+        public void UpdateProgress(string jobId, string taskId, int progress)
+        {
+            using (var activity = this._activitySource.StartActivity("update task progress"))
+            {
+                // activity.AddTag("progress", progress);
+
+                using (var content = new StringContent(JsonConvert.SerializeObject(new
+                {
+                    percentage = progress
+                }, this._jsonSerializerSettings)))
+                {
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                    Update(jobId, taskId, content);
+                }
+            }
+        }
+
+        public void UpdateCompletion(string jobId, string taskId)
+        {
+            using (this._activitySource.StartActivity("update task completed"))
+            {
+                using (var content = new StringContent(JsonConvert.SerializeObject(new
+                {
+                    percentage = 100,
+                    status = Status.COMPLETED
+                }, this._jsonSerializerSettings)))
+                {
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                    Update(jobId, taskId, content);
+                }
+            }
+        }
+
+        public void UpdateReject(string jobId, string taskId, int attempts, string reason, bool resettable)
+        {
+            using (var activity = this._activitySource.StartActivity("reject task"))
+            {
+                // activity.AddTag("attempts", attempts);
```

Comment:
remarks?

Reply:
This is because of an issue with the telemetry from when @cl-shaharshavit-rnd added otl.